rfcs/006-delia-replication.md

```diff
@@ -0,0 +1,113 @@
+# RFC-006: Automated Multi-Deal Submission in Delia for Replication
+
+**Author:** Pete — @pete-eiger  
+**Date:** 2025-02-24
+
+## Abstract
+
+This RFC proposes modifications to the Delia storage deal flow so that, instead of submitting a single deal proposal, the client automatically submits three identical deal proposals. By looping through the proposal process three times and executing the necessary RPC calls in parallel, the client not only streamlines the storage deal process but also increases data replication in the protocol. This replication is a core feature of the Polka Storage solution, ensuring that data is redundantly stored across multiple providers for improved availability, durability, and resilience in a decentralized environment. This change lays the groundwork for further enhancements such as automated deal acceptance and flexible provider selection.
+
+## Introduction
+
+Polka Storage relies on replicating data across multiple storage providers to ensure high availability and fault tolerance. Currently, Delia handles storage deals on a one-to-one basis: a client calculates a piece CID, proposes a deal, uploads a file, and then publishes the deal. For true replication, it is desirable for a client to automatically generate multiple identical deals—specifically three—so that the same piece of data is stored redundantly.
+
+This RFC outlines the necessary modifications to the client-side deal flow to support the automatic submission of three deals. The changes will involve:
+
+- Looping through the existing deal proposal flow three times.
+- Making parallel JSON‑RPC calls against individual storage provider servers.
+- Aggregating status feedback in the user interface.
+- Addressing provider selection: either automatically choosing three providers or allowing the user to manually select three.
+
+## Motivation
+
+- **Replication:**  
+  Ensuring that data is stored in multiple locations is a cornerstone of a decentralized storage system. Automatically submitting three identical deals increases redundancy and resilience, so that if one deal or provider fails, the data remains available elsewhere.
+
+- **Efficiency:**  
+  Submitting multiple deals automatically streamlines the process and reduces manual intervention, which is especially important in a decentralized system where network conditions and provider availability may vary.
```

Comment:
While this sentence isn't false, it's not relevant to the document as it's not part of the motivation behind adding replication. We could very well add manual replication which would be better (since it provides more control & information to the end-user) in cases where "availabilty may vary" as you put it.

In Portuguese we call this "encher chouriços"